### PR TITLE
feat(cli): Add upgrade command for CLI self-update

### PR DESCRIPTION
**Pull request by cameroncooke**
Adds a new top-level `xcodebuildmcp upgrade` command that compares the
installed version against the latest GitHub release, detects the install
method (Homebrew, npm-global, npx, or unknown), shows a truncated release
notes excerpt, and either runs the correct upgrade command with inherited
stdio or prints the manual command for unsupported channels.

## Behavior

- `--check` reports current vs. latest without prompting (CI-friendly).
- `--yes` / `-y` skips the confirmation and auto-runs the upgrade.
- Homebrew: `brew update` then `brew upgrade xcodebuildmcp`.
- npm-global: `npm install -g xcodebuildmcp@latest`.
- npx: never auto-runs — prints the ephemeral-install note and exits 0
  so `--yes` does not error in scripts.
- Unknown install: prints manual options, never auto-runs.
- Non-TTY callers that could auto-upgrade but omit `--yes` exit 1 so
  scripts do not hang on an invisible confirm.
- GitHub failures (timeout, non-200, 403 rate limit, malformed JSON) are
  reported with a clear message plus the manual upgrade command when
  detection succeeded.

## Implementation notes

- The command short-circuits in `src/cli.ts` mirroring `init` / `setup`,
  so no runtime/daemon bootstrap runs for this path.
- Version generator (`scripts/generate-version.ts`) now emits
  `packageName`, `repositoryOwner`, `repositoryName` parsed from
  `package.json`; the command reads these from `src/utils/version`
  instead of hardcoding identifiers. Generator fails the build if
  `repository.url` is not a parseable GitHub URL.
- Upgrade execution uses `spawn(..., { stdio: 'inherit' })` with
  SIGINT/SIGTERM forwarding; the buffered `CommandExecutor` is
  deliberately not used because package-manager output and prompts need
  to stream live.
- Semver comparison is done locally (leading `v` stripped, prerelease
  suffixes supported) to avoid adding a `semver` dependency for one
  command.
- Install detection is conservative: realpath-based on
  `process.argv[1]` and `process.execPath`, matching Homebrew Cellar
  paths (Intel and Apple Silicon), global `node_modules/xcodebuildmcp`,
  and the npx `_npx` cache. Unknown paths never auto-run.

## Tests

67 unit tests cover the full matrix: version parsing/compare edge cases,
detection for each channel with exact argv assertions, TTY and non-TTY
paths for every flag combination, release-notes truncation at both the
20-line and 2000-char caps, and every GitHub failure class. All
dependencies are injected through `UpgradeDependencies` — no real
network, spawn, or filesystem access in tests.

### CODE REVIEW

File: src/cli/commands/upgrade.ts [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+export interface UpgradeOptions {
+  check: boolean;
+  yes: boolean;
+}
+
+// --- Version comparison ---
+
+export function parseVersion(raw: string): ParsedVersion | undefined {
+  const stripped = raw.startsWith('v') ? raw.slice(1) : raw;
+  const match = stripped.match(/^(\d+)\.(\d+)\.(\d+)(?:-([a-zA-Z0-9.]+))?(?:\+[a-zA-Z0-9.]+)?$/);
```
[the remark this answers is not in the document]
Comment (cameroncooke) [reply src/cli/commands/upgrade.ts]: Fixed in 59d3e808. `parseVersion` regex now includes `-` in both the prerelease and build-metadata character classes, matching SemVer 2.0.0. Added tests for `1.0.0-alpha-1` and `1.0.0-rc-1+build-hash`.

File: src/cli/commands/upgrade.ts [hunk 2]
```diff
@@ -0,0 +1,841 @@
...
+    const reason = error instanceof Error ? error.message : String(error);
+    throw new Error(`couldn't determine latest version from Homebrew: ${reason}`);
+  }
+
+  let data: unknown;
+  try {
+    data = JSON.parse(result.stdout);
+  } catch {
+    throw new Error("couldn't determine latest version from Homebrew: invalid JSON output");
+  }
```
[the remark this answers is not in the document]
Comment (cameroncooke) [reply src/cli/commands/upgrade.ts:272]: Fixed in 59d3e808. Added an explicit `exitCode !== 0` check in `fetchLatestVersionFromHomebrew` before attempting `JSON.parse`, mirroring the npm sibling. The existing empty-`formulae` path is preserved because `brew info` quirkily exits 0 for missing formulas. Added a test for the non-zero exit path.

File: src/cli/commands/upgrade.ts [hunk 3]
```diff
@@ -0,0 +1,841 @@
...
+export interface UpgradeOptions {
+  check: boolean;
+  yes: boolean;
+}
+
+// --- Version comparison ---
+
+export function parseVersion(raw: string): ParsedVersion | undefined {
+  const stripped = raw.startsWith('v') ? raw.slice(1) : raw;
+  const match = stripped.match(/^(\d+)\.(\d+)\.(\d+)(?:-([a-zA-Z0-9.]+))?(?:\+[a-zA-Z0-9.]+)?$/);
```
[the remark this answers is not in the document]
Comment (cameroncooke) [reply src/cli/commands/upgrade.ts]: Fixed in 59d3e808 — same root cause as the Cursor Bugbot thread. Regex now allows `-` in both prerelease and build-metadata identifiers, with tests for hyphenated semver inputs.

File: src/cli/commands/upgrade.ts [hunk 4]
```diff
@@ -0,0 +1,847 @@
...
+    if (aIsNum && bIsNum) {
+      const diff = Number(aParts[i]) - Number(bParts[i]);
+      if (diff !== 0) return diff;
+      continue;
+    }
+
+    if (aIsNum) return -1;
+    if (bIsNum) return 1;
+
+    const cmp = aParts[i].localeCompare(bParts[i]);
```
[the remark this answers is not in the document]
Comment (cameroncooke) [reply src/cli/commands/upgrade.ts]: Fixed in b17e405a. Replaced `localeCompare` with a UTF-16 code-unit comparison, which matches the ASCII sort order SemVer 2.0.0 §11 requires. Added a test (`Alpha` vs `alpha`) that exercises the case where the two orderings diverge — verified manually that `localeCompare` returns 1 in common locales (en-US, tr-TR, de-DE) while ASCII correctly returns -1.
